lib/internal/Magento/Framework/DomDocument/DomDocumentFactory.php

```diff
@@ -3,20 +3,46 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+
 namespace Magento\Framework\DomDocument;
 
+use DOMDocument;
+
 /**
  * DOM document factory
  */
 class DomDocumentFactory
 {
     /**
+     * @var \Magento\Framework\ObjectManagerInterface
+     */
+    private $objectManager;
+
+    /**
+     * DomDocumentFactory constructor.
+     * @param \Magento\Framework\ObjectManagerInterface $objectManager
+     */
+    public function __construct(\Magento\Framework\ObjectManagerInterface $objectManager)
```

Comment:
Do we need to make the changes to this file or are we always using the new Dom factory class? My worry here is that a change like this will break backwards compatibility.

Reply:
I see, this breaks BC. I can make the `$objectManager` dependency optional?

Reply:
Technically this is a private class so we could choose to break BC in this case. Otherwise you could also make the object manager a static call. Another thing to note is that the factory should only be concerned with creating the object and not filling it. In this case `initializeDocument` in the other factory and calling `$dom->loadXML($data)` load in the other factory is not really the way to go in my mind.

Reply:
You're right. Factories should only instantiate objects. Wouldn't it be better to migrate all the domain specific code back to `\Magento\Developer\Model\XmlCatalog\Format\PhpStorm`?

Sorry for all the corrections, refactorings, etc. I'm learning a lot from creating Magento 2 pull requests :).

Reply:
No worries. I see two options here.

1. Migrate all the domain specific code back to the `\Magento\Developer\Model\XmlCatalog\Format\PhpStorm`.
2. Create our own classes for the two dom documents that have the public methods that we need. Then change the new factories to instantiate these new objects.

Not sure what you would prefer to do. If you do go with the idea of new classes then please have a look through http://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines/technical-guidelines.html#2-class-design and of course let me know if you need some assistance.

Reply:
@tdgroot do you think you will have time to look into this?

Reply:
@dmanners I'm currently too busy to get working on this PR. Next week I hope to find some time.